src/domain/flow-system/src/services/flow/flow_service.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use event_sourcing::LoadError;
+use internal_error::{ErrorIntoInternal, InternalError};
+use kamu_core::DatasetNotFoundError;
+use opendatafabric::{AccountID, AccountName, DatasetID};
+use tokio_stream::Stream;
+
+use crate::{DatasetFlowType, FlowID, FlowKey, FlowState, SystemFlowType};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait FlowService: Sync + Send {
+    /// Runs the update main loop
+    async fn run(&self) -> Result<(), InternalError>;
+
+    /// Creates a new manual flow request
+    async fn request_manual_flow(
```

Comment:
"flow request" sounds confusing as no "requests" are seen anywhere else in this sub-domain's terminology.

Perhaps a better name would be `create_or_trigger_manual_flow`?

Reply:
`trigger_manual_flow`